cmd/local-artifact-mirror/artifact.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/file"
+	"oras.land/oras-go/v2/registry"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/credentials"
+)
+
+// DockerConfig represents the content of the '.dockerconfigjson' secret.
+type DockerConfig struct {
+	Auths map[string]DockerConfigEntry `json:"auths"`
+}
+
+// DockerConfigEntry represents the content of the '.dockerconfigjson' secret.
+type DockerConfigEntry struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// registryAuth returns the authentication store to be used when reaching the
+// registry. The authentication store is read from the cluster secret named
+// 'registry-creds' in the 'kotsadm' namespace.
+func registryAuth(ctx context.Context) (credentials.Store, error) {
+	nsn := types.NamespacedName{Name: "registry-creds", Namespace: "kotsadm"}
+	var sct corev1.Secret
+	if err := kubecli.Get(ctx, nsn, &sct); err != nil {
+		if errors.IsNotFound(err) {
+			// if we can't locate a secret then returns an empty
+			// credentials store so we attempt to fetch the assets
+			// without authentication.
+			return credentials.NewMemoryStore(), nil
```

Comment:
should we log something indicating that we will pull anonymously?

Reply:
makes sense. done.